### ISSUE TITLE
Removal of ISO resources under https://standards.iso.org/ittf/PubliclyAvailableStandards/ISO_19139_Schemas

### ISSUE THREAD

**Issue by MarcoMinghini** [ISO-TC211/schemas#24]
Dear all,
on behalf of the INSPIRE team, I am writing to ask for some explanation of the recent removal of all the resources under https://standards.iso.org/ittf/PubliclyAvailableStandards/ISO_19139_Schemas. These include the schema http://standards.iso.org/ittf/PubliclyAvailableStandards/ISO_19139_Schemas/gmd/gmd.xsd, which is referenced in other widely-used schemas.

This breaking change is already causing issues in the INSPIRE validation community - see for example [[1](https://github.com/inspire-eu-validation/community/issues/245)], [[2](https://github.com/inspire-eu-validation/community/issues/246)], [[3](https://github.com/inspire-eu-validation/community/issues/247)], [[4](https://inspire.ec.europa.eu/forum/discussion/view/264812/iso-ap-100-schemalocation-to-iso-website-not-available)].

Can you please inform us whether this is just a temporary or a permanent change?
Thank you.

**Comment by ejbleys** [comment]
Dear Marco
I have no idea how the material is placed/removed from the http://standards.iso.org/ittf/PubliclyAvailableStandards/ <http://standards.iso.org/ittf/PubliclyAvailableStandards/> site
However I recognise that the ISO 19139 schemas should (but currently not) be available from https://schemas.isotc211.org/19139/-/gmd/ <https://schemas.isotc211.org/19139/-/gmd/>
I will endeavour to have suitable schemas placed in that location.
If someone has a full copy of the appropriate schemas could they be passed to me.
	- the copies I have may have been modified from the original.

Yours sincerely
Evert
Evert Bleys
Convenor ISO TC 211 XMG
4 Tudor Place HUGHES ACT 2605
Australia
+61 (0)2 62811773
+61 (0)411 483 876
ejbleys@gmail.com
Skype ejbleijs@gmail.com

**Comment by PeterParslow** [comment]
I suggest we (you Evert @ejbleys ) escalate the issue at standards.iso.org to Mats to take up with ISO CS.

Let's hope they don't have a policy of simply taking down these resources when the related standard is withdrawn!

**Comment by ronaldtse** [comment]
@ejbleys @PeterParslow since this is a matter of urgency, we can work on making those schemas available at the schemas.isotc211.org location once someone provides us with a copy of the schema. Just ping me!

**Comment by ronaldtse** [comment]
Guys, I found these on web.archive.org and also in the ISO Geodetic Registry code. Here they are.
[19139.zip](https://github.com/ISO-TC211/schemas/files/4317361/19139.zip)

**Comment by PeterParslow** [comment]
Be wary of which set we take on & host. ISO 19139 schemas have a long history of varied & incomplete schema sets, which have caused all sorts of "gotchas".

For the UK, we have summarised the position (as it was a couple of years ago!) here: https://www.agi.org.uk/40-gemini/1048-uk-gemini-encoding-guidance#2.1

Our UK preferred sets depend on the standards.iso.org set. The various OGC ones need different handling in the validators.

**Comment by ronaldtse** [comment]
@PeterParslow could you have a look at the zip archive uploaded? From its contents, it's likely the files were from the standards.iso.org site.

**Comment by ronaldtse** [comment]
I've made a pull request to resolve this in #25.

**Comment by PeterParslow** [comment]
A quick look suggests they might be:
 - contrary to the date in the readme, several of the files were updated in 2012 (the @version attribute of the root element). This fits in with XMG work under Jean. The issue then was that W3C had just published XLink 1.1 & OGC had decided to adopt it. See http://schemas.opengis.net/iso/19139/ReadMe.txt - which *may* be a better source.

- a number of the files contain relative paths in schemaLocations e.g. to import the GML schemas.

Of course, that second point means that they are not simply usable directly if placed on a different site.

**Comment by ronaldtse** [comment]
@PeterParslow thanks! If we do host them we will have to fix the import paths, but maybe the first point means we will need to seek another source.

web.archive.org does seem to contain this content fully. Let me download them and revert.

**Comment by ronaldtse** [comment]
@PeterParslow the copies from web.archive.org are now in #25. These are supposed to be identical to the ones from standards.iso.org. They were indeed, very different.

**Comment by PeterParslow** [comment]
The resources are also required in their original authoritative place, unless ISO (or TC211) can manage a rapid communication process to persuade lots of implementations to change where they look for them.

For example the aforementioned UK GEMINI references a number of code lists that should reside in http://standards.iso.org/ittf/PubliclyAvailableStandards/ISO_19139_Schemas/resources/codelist/               gmxCodelists.xml

This path not only exists in approximately 25,000 metadata records, but it does so because it is implemented in at least five different software systems used by hundreds of organisations.

**Comment by PeterParslow** [comment]
Looking at https://standards.iso.org/ittf/PubliclyAvailableStandards/ the problem seems to have been caused by ISO's migration to a new content management system - the web pages now have the same look and feel as the new document sharing system (for which the 'top level' of TC211 is part of the pilot).

I guess they haven't created places in the new system for withdrawn/superseded standards, or it could be that they have only migrated ISO/IEC standards i.e. those which come under JTC1 (although they haven't got all of those). 

I still reckon it's more important to get ISO to reinstate them.

I expect we have common cause to complain alongside other JTC1 committees - although perhaps we're the only one that takes the lifecycle management of our resources at all seriously!

**Comment by ejbleys** [comment]
Hi Guys
I recall that at the Plenary in Omiya, Knut mentioned that all standards will be remain available.
	Hence there is no need to specifically develop a location for <withdrawn | superseded>
		the structure https://schemas.isotc211.org/<standardNo.>/-<partNo.>/<nameSpaceAbbreviation>/<versionNo.> <https://schemas.isotc211.org/%3CstandardNo.%3E/-%3CpartNo.%3E/%3CnameSpaceAbbreviation%3E/%3CversionNo.%3E> can already deal with that
	BUT It would have been unreasonable to make a commitment for them to always remaining in the same location, the world moves on.
I recommend that ISO TC 211 consider taking on the tasks to bring all (available) schemas into the https://schemas.isotc211.org/ <https://schemas.isotc211.org/> space
There will need to be some realignment of schemaLocation=“”, noting that the next time we need to move URL root, we’ll have to play that game again.

It would be nice for the material to <always remain in | be returned to>  standards.iso.org/ittf/PubliclyAvailableStandards/ <http://standards.iso.org/ittf/PubliclyAvailableStandards/> but I note that it has moved from http to https so the URL has already changed

Cheers
Evert

Evert Bleys
4 Tudor Place HUGHES ACT 2605
Australia
+61 (0)2 62811773
+61 (0)411 483 876
ejbleys@gmail.com
Skype ejbleijs@gmail.com

**Comment by ronaldtse** [comment]
@ejbleys agree. According to the given URL pattern, what should the proper path be for the URL in the original question: `http://standards.iso.org/ittf/PubliclyAvailableStandards/ISO_19139_Schemas/gmd/gmd.xsd`?

The agreed pattern is: ` https://schemas.isotc211.org/<standardNo.>/<partNo.>/<nameSpaceAbbreviation>/<versionNo.>`

Would it be `https://schemas.isotc211.org/19139/-/gmd/gmd.xsd`? There are 2 issues:

* We can't differentiate future 19139 editions if there will be any (maybe not in this case since 19139 is split into parts, but still).
* There is no `<versionNo.>` in the `gmd/gmd.xsd` part.

**Comment by ejbleys** [comment]
Thanks Ron
Opinions please
To fit the pattern properly it would be https://schemas.isotc211.org/19139/-/gmd/1.0/gmd.xsd <https://schemas.isotc211.org/19139/-/gmd/1.0/gmd.xsd> 
If there were to be a new version of ISO 19139 then it would get a part number (eg -1, -2, -n) 
You are correct in suggesting that the pattern was considered after 19139 was developed
	but we should start with a version now 
		logically it should be 0.0 but I think that does not make a lot of sense in the schema of things
We would leave the namespace URL the same to limit the fallout from the changes

I have looked through the files from the web.archive
some files need adjusting as some refer to: 
	itty location - change to either 
			relative path (can be checked prior to move); or 
			https://schemas.isotc211.org/ <https://schemas.isotc211.org/>… (can only be checked after being transferred to that location 
		I lean towards the relative path (validation possible), but the full URL is neater
	xlink via relative path "../../../xlink/xlink.xsd” - change to https://www.w3.org/1999/xlink.xsd <https://www.w3.org/1999/xlink.xsd>

Cheers
E

Evert Bleys
4 Tudor Place HUGHES ACT 2605
Australia
+61 (0)2 62811773
+61 (0)411 483 876
ejbleys@gmail.com
Skype ejbleijs@gmail.com

**Comment by ronaldtse** [comment]
@ejbleys agree with `https://schemas.isotc211.org/19139/-/gmd/1.0/gmd.xsd`. I think using `1.0`  instead of `0.0` makes sense, since this is a migration -- people need to update their locations anyway.

Re: `xlink.xsd`: where are you seeing the relative paths to `xlink.xsd`? I did a search across all schema files and they're all using `https://www.w3.org/1999/xlink.xsd`. Maybe I'm missing something?

It is however true that we need to fix some schemaLocation paths which are pointing to non-existing ones on standards.iso.org. Filing new task.

**Comment by ejbleys** [comment]
Thanks
Yes but 19139 links to 19136 there in lies the issue
Unravelling the spaghetti - you up for a challenge?
Evert
Evert Bleys
4 Tudor Place HUGHES ACT 2605
Australia
+61 (0)2 62811773
+61 (0)411 483 876
ejbleys@gmail.com
Skype ejbleijs@gmail.com

**Comment by PeterParslow** [comment]
I may have over stated the size of the user problem. If ISO won’t put them back (or can’t be trusted!) I think we “just” need INSPIRE and OGC to change their top level files to import from our new location:

1.	The OGC CSW 2.0.2 schemas import them from there: http://schemas.opengis.net/csw/2.0.2/profiles/apiso/1.0.0/apiso.xsd (last update 2018-03-01)
2.	The INSPIRE set, constructed to bring together our official “dataset” metadata schemas & some unofficial “service” (ISO 19119) ones: http://inspire.ec.europa.eu/draft-schemas/inspire-md-schemas/apiso-inspire/apiso-inspire.xsd (last update 2010)

But I've only checked EU & UK advice on where to find the schemas.

**Comment by nmtoken** [comment]
schemaLocation in XML instances don't need to change, it just represents a hint to applied schema

**Comment by nmtoken** [comment]
schemaLocation in schema like http://schemas.opengis.net/csw/2.0.2/profiles/apiso/1.0.0/apiso.xsd and http://inspire.ec.europa.eu/draft-schemas/inspire-md-schemas/apiso-inspire/apiso-inspire.xsd will need to change; currently they import/include schema from http://standards.iso.org/ittf/PubliclyAvailableStandards

**Comment by fabiovin** [comment]
For the INSPIRE Reference Validator, we created a new version of the apiso schema (http://inspire.ec.europa.eu/draft-schemas/inspire-md-schemas-temp/apiso-inspire/apiso-inspire.xsd), to overcome the http/https redirect problem, importing the two schemas (gmd and gmx) from the OpenGIS repository.

It is used as fixed schema to validate the metadata and it work fine.

Looking at this ReadMe file (http://schemas.opengis.net/iso/19139/ReadMe.txt) it seems that files in the folder "iso/19139/20070417" should be the same as those were available at http://standards.iso.org/ittf/PubliclyAvailableStandards/ISO_19139_Schemas/.

So, I don't know if we can refer to them!

**Comment by ronaldtse** [comment]
@ejbleys Always up for spaghetti -- as long as there is an agreed upon end! 👍 

> Unravelling the spaghetti - you up for a challenge?

**Comment by MartinePaepen** [comment]
Hi,
we are implementing ISO 19115-2:2019-01.
And face the same problem:
the spec indicates that the online schema is availalbe on https://standards.iso.org/iso/19115/-2/gmi/
however if we use these, our INSPIRE metadata files doesn't validate due to the missing schema locations at http://standards.iso.org/ittf/PubliclyAvailableStandards/ISO_19139_Schemas/

For  us it is necessary to solve this quickly. Any idea how to cope with this in the short term?

Many Thanks!
Martine Paepen (VITO / Belgium)

**Comment by ronaldtse** [comment]
@MartinePaepen if you need to solve this quickly, the identical schema files of http://standards.iso.org/ittf/PubliclyAvailableStandards/ISO_19139_Schemas/ are already provided on schemas.isotc211.org  (since 842920a533e7b8ddd8a95c95a6a3bb6f373f6ace).

* If we change the absolute paths of http://standards.iso.org/ittf/PubliclyAvailableStandards/ISO_19139_Schemas/ in https://standards.iso.org/iso/19115/-2/gmi/1.0/*.xsd to a relative path
* You change to use https://schemas.isotc211.org/schemas/19115/-2/gmi/

Then the schemas should resolve. Thoughts @ejbleys ?

**Comment by MartinePaepen** [comment]
Trying to use  https://schemas.isotc211.org/schemas/19115/-2/gmi/ and I receive the following error:

404 Not Found
Code: NoSuchKey
Message: The specified key does not exist.
Key: 404.html
RequestId: 1ABF645DCD1175B5
HostId: tZC75zYXGyEemr4VckxZGQXZqO7KGPyVkkOP7yO4WF0m0OPfPAxDhfTcAgWqx8Pp5krZYA8fFXw=
An Error Occurred While Attempting to Retrieve a Custom Error Document
Code: NoSuchKey
Message: The specified key does not exist.
Key: 404.html

**Comment by ronaldtse** [comment]
@MartinePaepen The paths have to be fully specified: e.g.
https://schemas.isotc211.org/schemas/19115/-2/gmi/1.0/gmi.xsd

However, we haven't updated the `schemaLocations` yet of these files; need to confirm with @ejbleys who is the XMG convener.

**Comment by ronaldtse** [comment]
@MartinePaepen @ejbleys a pull request has been made in #30 , if this is agreed we can merge it -- then @MartinePaepen you can utilize the new location.

**Comment by ejbleys** [comment]
If you are truly implementing ISO 19115-2:2019 you should be using ISO 19115-3 not gmi
There is a plan (not fully agreed on) to break up ISO 19115-3 into:
	ISO 19115-1; ISO 19115-2; ISO 19103 (most of gco/gcx), and others as needed

Cheers
E
Evert Bleys
4 Tudor Place
HUGHES ACT 2605
Australia
email:  ejbleys@gmail.com
Mob:   +61 (0)411 483 876
Land:  +61 (0)2 6281 1773
Skype: ejbleijs@gmail.com

**Comment by ronaldtse** [comment]
@MartinePaepen : FYI #30 has been merged by @ejbleys , so https://schemas.isotc211.org/schemas/19115/-2/gmi/1.0/gmi.xsd (and other 19115-2) will refer to valid 19139 schemaLocations.

Closing this issue as the 19139 schemas are now available through schemas.isotc211.org courtesy of XMG.

**Comment by MartinePaepen** [comment]
If you are truly implementing ISO 19115-2:2019 you should be using ISO 19115-3 not gmi
There is a plan (not fully agreed on) to break up ISO 19115-3 into:
	ISO 19115-1; ISO 19115-2; ISO 19103 (most of gco/gcx), and others as needed

==> Is this true?

In the SPECs ISO19115-2:2019(E) p. 56 we find the link to https://schemas.isotc211.org/schemas/19115/-2/gmi/  However it states that the vesion number is 2.0 and this version is not available. 
And I see that the schema's still refer to https://standards.iso.org/ittf/PubliclyAvailableStandards/ISO_19139_Schemas/gmd/gmd.xsd which are not available.

**Comment by MartinePaepen** [comment]
@ejbleijs I have checked 19115-3 and it doesn't contain gmi and that is exactly what we need to put the acquisitionInformation and band information etc....

**Comment by MartinePaepen** [comment]
I have also checked the schemas of https://standards.iso.org/iso/19139/Schemas/ and also there is no gmi available.

**Comment by ejbleys** [comment]
> the GitHub repository that feeds https://schemas.isotc211.org/schemas/19115/-2/gmi/ <https://schemas.isotc211.org/schemas/19115/-2/gmi/> has been updated today in an attempt to overcome the issue
> 
> It now references https://schemas.isotc211.org/19139/-/gmd/1 <https://schemas.isotc211.org/19139/-/gmd/1>.0
GMI/2.0 has been superseded in ISO 19115-3 and is now referenced as MAC; MRC; MRL; and MSR
	The ISO/TC 211 XMG is attempting to herd-the-cats so that only once a schemas is accepted for release will the Annex describing it identify its intended location 
I appreciate the intrinsic difficulty caused by the Printed/Printable version of ISO 19115-2 not having been updated
The ISO/TC 211 XMG will (hopefully) work towards a stabilisation of the directory structure under the guidance provided by the resolution about directory structures from the Omiya Plenary.
In future the XMG will also attempt to undertaking that schemas will not be removed from the https://schemas.isotc211.org/ <https://schemas.isotc211.org/> site.

As we have no authority over the http(s)://standards.iso.org/iso <http://standards.iso.org/iso> and http(s)://standards.iso.org/ittf/PubliclyAvailableStandards/ <https://standards.iso.org/ittf/PubliclyAvailableStandards/> the same undertaking cannot be made for those repositories.

Yours faithfully

Evert

Evert Bleys
4 Tudor Place
HUGHES ACT 2605
Australia
email:  ejbleys@gmail.com
Mob:   +61 (0)411 483 876
Land:  +61 (0)2 6281 1773
Skype: ejbleijs@gmail.com

**Comment by ejbleys** [comment]
ISO 19115-2 was not part of ISO 19139
Evert Bleys
4 Tudor Place
HUGHES ACT 2605
Australia
email:  ejbleys@gmail.com
Mob:   +61 (0)411 483 876
Land:  +61 (0)2 6281 1773
Skype: ejbleijs@gmail.com

**Comment by ejbleys** [comment]
GMI deprecated in ISO 19115-3

Evert Bleys
4 Tudor Place
HUGHES ACT 2605
Australia
email:  ejbleys@gmail.com
Mob:   +61 (0)411 483 876
Land:  +61 (0)2 6281 1773
Skype: ejbleijs@gmail.com

**Comment by PeterParslow** [comment]
Marine, that second post is easier to answer: "gmi" wasn't part of ISO 19139; it was introduced in ISO 19139-2.

As specified in that standard, the gmi files live at https://www.isotc211.org/2005/gmi/.

Martine, do bear with Ron & Evert as they attempt to sweep up after ISO unexpectedly removing the schema set that most of us used!

**Comment by MartinePaepen** [comment]
Thanks @PeterParslow .
We were also using the  https://www.isotc211.org/2005/gmi/ schemas. However we see that they are not completely in line with the SPECs of ISO19115-2:2019(E). For example: MI_Operation contains in the spec parentOperation as Mandatory but is not part of the 2005/gmi MI_Operations
==> this makes it confusing and difficult to implement.

**Comment by PeterParslow** [comment]
@MartinePaepen . 19115-2 isn't a standard I have experience of. If you think the issue you've spotted is a problem in the standard document, please log it at http://isotc211.standardstracker.org/ - unless it's one of the issues already raised there (you can browse by standard number).

If you think the standard is correct & the schema isn't, then @ejbleys may prefer a separate GitHub issue for that.
